stl/inc/memory

```diff
@@ -393,6 +770,95 @@ void uninitialized_value_construct(const _NoThrowFwdIt _First, const _NoThrowFwd
     }
 }
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // clang-format off
+    template <_No_throw_forward_iterator _It, _No_throw_sentinel_for<_It> _Se>
+            requires default_initializable<iter_value_t<_It>>
+    _It _Zero_range(const _It _First, const _Se _Last) {
+        // clang-format on
+        // fill [_First, _Last) with zeroes
+        if constexpr (is_same_v<_Se, _It>) {
+            char* const _First_ch = reinterpret_cast<char*>(_First);
+            char* const _Last_ch  = reinterpret_cast<char*>(_Last);
+            _CSTD memset(_First_ch, 0, static_cast<size_t>(_Last_ch - _First_ch));
+            return _Last;
+        } else if constexpr (sized_sentinel_for<_Se, _It>) {
+            const _It _Final      = _RANGES next(_First, _Last - _First);
+            char* const _First_ch = reinterpret_cast<char*>(_First);
+            char* const _Final_ch = reinterpret_cast<char*>(_Final);
+            _CSTD memset(_First_ch, 0, static_cast<size_t>(_Final_ch - _First_ch));
+            return _Final;
+        } else {
+            const _It _Final      = _RANGES next(_First, _STD move(_Last));
+            char* const _First_ch = reinterpret_cast<char*>(_First);
+            char* const _Final_ch = reinterpret_cast<char*>(_Final);
+            _CSTD memset(_First_ch, 0, static_cast<size_t>(_Final_ch - _First_ch));
+            return _Final;

```

Comment:
Ditto "It's probably not worth the hassle to maintain more lines of code when this case should optimize to the same code as the two above."

Reply:
Ditto removed